security/tinc/src/opnsense/scripts/OPNsense/Tinc/tincd.py

```diff
@@ -107,7 +110,11 @@ def deploy(config_filename):
 if len(sys.argv) > 1:
     if sys.argv[1] == 'stop':
         for instance in glob.glob('/usr/local/etc/tinc/*'):
+            proc_cat = subprocess.Popen(['cat', '%s/tinc.conf' % instance], stdout=subprocess.PIPE)
```

Comment:
no need for subprocess here:
```
if os.path.exists('%s/tinc.conf' % instance):
    interface_name  = open('%s/tinc.conf' % instance).read().split('Device=')[-1].split()[0].split('/')[-1]
    if interface_name.startswith('tinc'):
        subprocess.run(['/sbin/ifconfig',interface_name,'destroy'])
```

basically does the same, a regexp could also work. although it's not 100% foolproof, it should be safe enough to use

Reply:
Implemented in https://github.com/opnsense/plugins/pull/1733/commits/b2238a760f6b45a689b2842af3f80055019adf9c